tests/rptest/tests/data_transforms_test.py

```diff
@@ -0,0 +1,127 @@
+# Copyright 2023 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+from rptest.clients.rpk import RpkTool
+from rptest.services.cluster import cluster
+from ducktape.utils.util import wait_until
+
+from rptest.tests.redpanda_test import RedpandaTest
+from rptest.clients.types import TopicSpec
+
+
+class WasmException(Exception):
+    def __init__(self, message):
+        self.message = message
+
+    def __str__(self):
+        return repr(self.message)
+
+
+class DataTransformsTest(RedpandaTest):
+    """
+    Tests related to WebAssembly powered data transforms
+    """
+    topics = [TopicSpec(partition_count=9), TopicSpec(partition_count=9)]
+
+    def __init__(self, test_context):
+        super(DataTransformsTest, self).__init__(test_context=test_context)
+        self._ctx = test_context
+        self._rpk = RpkTool(self.redpanda)
+
+    def _enable_wasm(self):
+        """
+        Enable data transforms and restart the cluster.
+        """
+        self._rpk.cluster_config_set("data_transforms_enabled", True)
+        self.redpanda.rolling_restart_nodes(self.redpanda.nodes)
+
```

Comment:
suggestion: Since you always want this config item enabled, you can actually pass it to the `RedpandaTest` constructor as an `extra_rp_conf` option:

```suggestion
    def __init__(self, test_context):
        super(DataTransformsTest, self).__init__(test_context=test_context,
                                                                              extra_rp_conf={'data_transforms_enabled': True})
        self._ctx = test_context
        self._rpk = RpkTool(self.redpanda)
```

This allows you to avoid the rolling restart

Reply:
Amazing, thanks!